File: python/reverse_integer.py
import math

class Solution(object):
    # o(1) time | O(1) space
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        MIN = -2147483648
        MAX = 2147483647
        res = 0
        while x != 0:
            digit = int(math.fmod(x, 10))
            x = int(x / 10)

            test = 10 * res + digit
            if test < MIN or test > MAX:
                return 0
            res = test
        return res
    

solution = Solution()
print(solution.reverse(-123))

File: src/main/java/june_2025/ReverseInteger.java
package june_2025;

public class ReverseInteger {

    // O(1) time | O(1) space
    public int reverse(int x) {
        int res = 0;
        while (x != 0) {
            int digit = x % 10;
            x = x / 10;
            long test = 10l * res + digit;
            if (test < Integer.MIN_VALUE || test > Integer.MAX_VALUE) {
                return 0;
            }
            res = (int) test;
        }
        return res;
    }

}
